src/rluni/robot/RWIP.py

```diff
@@ -223,6 +228,37 @@ async def control_loop(self, shutdown_event):
             loop_delay = self.precise_delay_until(end_time)
             loop_period = time.time() - loop_start_time
 
+    def precise_delay_until(self, end_time):
```

Comment:
Can we move this method to a different file? This would be useful as a library function or something

Reply:
Sure, this has been in here before, I only moved it. I recommend moving it as a seperate PR or as part of some of the refactoring you are doing